sys-boot/udk/udk-2018.ebuild

```diff
@@ -0,0 +1,256 @@
+
+if [[ ${PV} == *9999* ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/tianocore/edk2.git"
+	KEYWORDS=""
+else
+	MY_V="UDK$(ver_cut 1)"
+	SRC_URI="https://github.com/tianocore/edk2/archive/v${MY_V}.tar.gz -> ${P}.tar.gz
+	doc? ( https://github.com/tianocore/edk2/releases/download/v${MY_V}/${MY_V}.Documents.zip -> ${P}-docs.zip )"
+	S="${WORKDIR}/edk2-v${MY_V}"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+LICENSE="BSD-2"
+SLOT="0"
+IUSE="doc examples"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+RESTRICT="strip"
+
+RDEPEND="${PYTHON_DEPS}"
+DEPEND="${RDEPEND}
+	>=dev-lang/nasm-2.14.02
+	>=sys-power/iasl-20160729
+	doc? ( app-arch/unzip )"
+
+DOCDIR="${WORKDIR}/Documents"
+HTML_DOCS="${DOCDIR}"/*
+
+# Generated libs for EFI can contain WX sections
+QA_EXECSTACK="
+	usr/lib*/libBaseLib.a:*
+	usr/lib*/libBaseIoLibIntrinsic.a:*
+"
+
+pkg_setup() {
+	python_setup 'python2.7'
+
+	local uname_arch=$(uname -m | sed -e 's:i[3456789]86:IA32:')
+	if [[ ${uname_arch} == "x86_64" ]] || [[ ${uname_arch} == "amd64" ]] ; then
+		ARCH=X64
+	else
+		ARCH=${uname_arch}
+	fi
+
+	# Select toolchain within predefined ones
+	if tc-is-gcc; then
+		if ver_test $(gcc-version) -ge 5; then
+			TOOLCHAIN_TAG="GCC5"
+		elif ver_test $(gcc-version) -ge 4.9; then
+			TOOLCHAIN_TAG="GCC49"
+		elif ver_test $(gcc-version) -ge 4.8; then
+			TOOLCHAIN_TAG="GCC48"
+		elif ver_test $(gcc-version) -ge 4.7; then
+			TOOLCHAIN_TAG="GCC47"
+		elif ver_test $(gcc-version) -ge 4.6; then
+			TOOLCHAIN_TAG="GCC46"
+		elif ver_test $(gcc-version) -ge 4.5; then
+			TOOLCHAIN_TAG="GCC45"
+		elif ver_test $(gcc-version) -ge 4.4; then
+			TOOLCHAIN_TAG="GCC44"
+		fi
+	elif tc-is-clang; then
+		if ver_test $(clang-version) -ge 3.8; then
+			TOOLCHAIN_TAG="CLANG38"
+		elif ver_test $(clang-version) -ge 3.5; then
+			TOOLCHAIN_TAG="CLANG35"
+		fi
+	fi
+	if [[ -z ${TOOLCHAIN_TAG} ]]; then
+		TOOLCHAIN_TAG="ELFGCC"
+	fi
+}
+
+src_unpack() {
+	unpack ${A}
+
+	local doc_name
+	local f
+	if use doc; then
+		rm -f "${DOCDIR}/"*.chm || die
+		pushd "${DOCDIR}" >/dev/null || die
+		for f in "${DOCDIR}/"*.zip; do
+			doc_name=$(basename "${f}" .zip)
+			unpack "${f}"
+			mv "${DOCDIR}/html" "${DOCDIR}/${doc_name}" || die
+		done
+		popd >/dev/null || die
+		rm -f "${DOCDIR}/"*.zip || die
+	fi
+}
+
+src_configure() {
+	# Compile of Base Tools is required for further setting up the environment
+	# Base tools does not like parallel make
+	local cflags_save=${CFLAGS}
+	append-cflags -MD -fshort-wchar -fno-strict-aliasing
+	append-cflags -nostdlib -c -fPIC
+	sed -e "s:^\(BUILD_CFLAGS\s*=\).*$:\1 ${CFLAGS}:" \
+		-i "${S}/BaseTools/Source/C/Makefiles/header.makefile" \
+		|| die "Failed to update makefile header"
+	CFLAGS=${cflags_save}
+	local make_flags=(
+		BUILD_CC="$(tc-getBUILD_CC)"
+		BUILD_CXX="$(tc-getBUILD_CXX)"
+		BUILD_AS="$(tc-getBUILD_AS)"
+		BUILD_AR="$(tc-getBUILD_AR)"
+		BUILD_LD="$(tc-getBUILD_LD)"
+	)
+	emake "${make_flags[@]}" -j1 -C BaseTools
+	. edksetup.sh
```

Comment:
That sounds like a very bad idea. Why do you need to do this?

Reply:
This is the way the upstream package work. One need to build the base tools, then execute setup in order to add environment variables, and finally, compile the project, using tools build at step 1.

Reply:
Do you need those vars for anything except `build` command below? If not, move it closer to it, and wrap both in subshell `( ... )` to avoid those vars colliding with package manager internals.

Reply:
The `edksetup` command is creating the configuration file which is adapted just after the call to this command line. I could eventually move the command and the following `sed`s to the `src_compile` function, but I am not sure that it would respect the _ebuild spirit_.